x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java

```diff
@@ -514,6 +529,81 @@ public void testLicenseUpdateFailureHandlerUpdate() throws Exception {
         }
     }
 
+    public void testSecurityHandlerIsAlwaysInstalled() throws IllegalAccessException {
+        final Logger amLogger = LogManager.getLogger(ActionModule.class);
+        Loggers.setLevel(amLogger, Level.DEBUG);
+        final MockLogAppender appender = new MockLogAppender();
+        Loggers.addAppender(amLogger, appender);
+        appender.start();
+
+        Settings settings = Settings.builder()
+            .put("xpack.security.enabled", false)
+            .put("path.home", createTempDir())
+            .build();
+        SettingsModule settingsModule = new SettingsModule(Settings.EMPTY);
+        ThreadPool threadPool = new TestThreadPool(getTestName());
+
+        try {
+            UsageService usageService = new UsageService();
+            Security security = new Security(settings, null);
+
+            appender.addExpectation(new MockLogAppender.SeenEventExpectation(
+                "Security rest wrapper", ActionModule.class.getName(), Level.DEBUG,
+                "Using REST wrapper from plugin org.elasticsearch.xpack.security.Security"
+            ));
```

Comment:
Nit: This logging line is the first thing the code does when it sees a rest wrapper. So strictly speaking, at this point, it is still possible that the rest wrapper will be rejected later. It is not really an issue right now, because the code throws if the rest wrapper is not installed and the test would fail. But it does not reads too well for me.

Reply:
I have added some comments to the test